### PR TITLE
[tests-only][full-ci] Add PROPFIND tests for public link endpoint

### PR DESCRIPTION
**Pull request by saw-jan**
## Related Issue
- https://github.com/owncloud/core/issues/39707

## How Has This Been Tested?
- test environment: local

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiWebdavOperations/propfind.feature [hunk 1]
```diff
@@ -47,4 +47,41 @@ Feature: PROPFIND
     Examples:
       | dav_path                         | depth_infinity_allowed | depth    | http_status |
       | /remote.php/dav/spaces/%spaceid% | 1                      | 0        | 207         |
-      | /remote.php/dav/spaces/%spaceid% | 1                      | infinity | 207         |
\ No newline at end of file
+      | /remote.php/dav/spaces/%spaceid% | 1                      | infinity | 207         |
+
+
+  Scenario: send PROPFIND requests to public link
```
Comment (SagarGi) [inline tests/acceptance/features/apiWebdavOperations/propfind.feature:53]:
```suggestion
  Scenario: send PROPFIND request to a public link
```

File: tests/acceptance/features/apiWebdavOperations/propfind.feature [hunk 2]
```diff
@@ -47,4 +47,41 @@ Feature: PROPFIND
...
+    And user "Alice" has created a public link share with settings
+      | path        | /PARENT |
+      | permissions | read    |
+    When the public sends "PROPFIND" request to the last public link share using the new public WebDAV API
+    Then the HTTP status code should be "207"
+    And the value of the item "//d:href" in the response should match "/%base_path%\/remote.php\/dav\/public-files\/%public_token%\/$/"
+    And the value of the item "//oc:public-link-share-owner" in the response should be "Alice"
+
+
+  Scenario: send PROPFIND requests to public link shared with password
```
Comment (SagarGi) [inline tests/acceptance/features/apiWebdavOperations/propfind.feature:65]:
```suggestion
  Scenario: send PROPFIND request to a public link shared with password
```

File: tests/acceptance/features/apiWebdavOperations/propfind.feature [hunk 3]
```diff
@@ -47,4 +47,41 @@ Feature: PROPFIND
...
+      | path        | /PARENT |
+      | permissions | read    |
+      | password    | 1111    |
+    When the public sends "PROPFIND" request to the last public link share using the new public WebDAV API with password "1111"
+    Then the HTTP status code should be "207"
+    And the value of the item "//d:href" in the response should match "/%base_path%\/remote.php\/dav\/public-files\/%public_token%\/$/"
+    And the value of the item "//oc:public-link-share-owner" in the response should be "Alice"
+
+
+  Scenario: send PROPFIND requests to public link shared with password (request without password)
```
Comment (SagarGi) [inline tests/acceptance/features/apiWebdavOperations/propfind.feature:78]:
```suggestion
  Scenario: send PROPFIND request to a public link shared with password (request without password)
```

Comment (SagarGi) [inline tests/acceptance/features/apiWebdavOperations/propfind.feature:78]: can we make a scenarios like this but with incorrect password?

File: tests/acceptance/features/apiWebdavOperations/propfind.feature [hunk 4]
```diff
@@ -47,4 +47,41 @@ Feature: PROPFIND
     Examples:
       | dav_path                         | depth_infinity_allowed | depth    | http_status |
       | /remote.php/dav/spaces/%spaceid% | 1                      | 0        | 207         |
-      | /remote.php/dav/spaces/%spaceid% | 1                      | infinity | 207         |
\ No newline at end of file
+      | /remote.php/dav/spaces/%spaceid% | 1                      | infinity | 207         |
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiWebdavOperations/propfind.feature:50]: i think depth= infinite was not allowed

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 5]
```diff
@@ -1658,4 +1658,56 @@ public function setUpScenario(BeforeScenarioScope $scope):void {
 		$this->featureContext = $environment->getContext('FeatureContext');
 		$this->occContext = $environment->getContext('OccContext');
 	}
+
+	/**
+	 * @When /^the public sends "([^"]*)" request to the last public link share using the (old|new) public WebDAV API$/
+	 * @When /^the public sends "([^"]*)" request to the last public link share using the (old|new) public WebDAV API with password "([^"]*)"$/
```
Comment (amrita-shrestha) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:1664]: can't we use regex and make this step one

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 6]
```diff
@@ -1658,4 +1658,56 @@ public function setUpScenario(BeforeScenarioScope $scope):void {
...
+					<oc:public-link-expiration/>
+					<oc:public-link-share-datetime/>
+					<oc:public-link-share-owner/>
+				</d:prop>
+			</d:propfind>';
+		}
+		$token = $this->featureContext->getLastPublicShareToken();
+		$davPath = WebDavHelper::getDavPath(
+			null,
+			0,
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:1693]: `0` is not a DAV version. maybe providing null is more dynamic. the `getDavPath` func. will determine by itself.

File: tests/acceptance/features/apiWebdavOperations/propfind.feature (hunk 4 above)
Comment (saw-jan) [reply tests/acceptance/features/apiWebdavOperations/propfind.feature:50]: That's skipped on oc10

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

**Comment by saw-jan** [comment]
Do we want to add these tests to oCIS as well?

CC @phil-davis

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Comment by phil-davis** [comment]
Rebased to get fresh CI

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
> Do we want to add these tests to oCIS as well?

@saw-jan yes, it will be good to know how this behaves on ocis. I raised issue https://github.com/owncloud/ocis/issues/5336 so that can be tracked and done.

**Pull request merged**